core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2504,7 +2504,7 @@ CompletableFuture<Void> writeShareGroupState(List<PersisterStateBatch> stateBatc
                 .setGroupId(this.groupId)
                 .setTopicsData(List.of(new TopicData<>(topicIdPartition.topicId(),
                     List.of(PartitionFactory.newPartitionStateBatchData(
-                        topicIdPartition.partition(), stateEpoch, startOffset(), leaderEpoch, stateBatches))))
+                        topicIdPartition.partition(), stateEpoch, startOffset(), inFlightTerminalRecords(), leaderEpoch, stateBatches))))
```

Comment:
I know that `inFlightTerminalRecords` was the previous name before the KIP was approved, but it's changed now. Please can we expunge the old name for the sake of the future maintainers.